app/scripts/controllers/network/network-controller.test.js

```diff
@@ -157,6 +168,8 @@ describe('NetworkController', () => {
     describe('#setProviderType', () => {
       it('should update provider.type', async () => {
         await withController(async ({ controller }) => {
+          setupMockRpcBlockResponses();
```

Comment:
I feel like we could be a bit more explicit in these tests when we mock the RPC requests. Taking this test as an example, it seems a bit inconsistent that we are specifying that we want to mock mainnet on line 174 but not specifying the network type on line 173. I realize that the default network type for this function is localhost, but I feel like one reason this function has a default is because NetworkController has the same default. In this case and similar cases, it seems like we shouldn't rely on this presumption. I think long-term NetworkController should require consumers to explicitly specify which kind of network they want to connect to.

What do you think about?

``` suggestion
        await withController(
          {
            provider: {
              type: NETWORK_TYPES.RPC,
              // ... anything else that is required ...
            }
          },
          async ({ controller }) => {
            setupMockRpcBlockResponses({ networkType: NETWORK_TYPES.RPC });
            setupMockRpcBlockResponses({ networkType: NETWORK_TYPES.MAINNET });
            await controller.initializeProvider();
            controller.setProviderType('mainnet');
            const { type } = controller.getProviderConfig();
            expect(type).toStrictEqual('mainnet');
          });
        )
```

Reply:
I was trying to make these setup helpers concise, so they don't distract from the purpose of each test. i.e. does it matter what the default is?

Maybe it does in this example actually, to show that the default is different than the network we're switching to. But in other cases I'm not so sure.

I can at least update `setupMockRpcBlockResponses` to explicitly reference the default network of the network controller, so it's clear why that is the default and so that it remains in-sync.

Reply:
I've implemented my suggestion here: 055ec6ab3fe3ed4282696c474bed47ec8005b5d9
Let me know if that makes sense